code/test_bsg.c

```diff
@@ -2,8 +2,50 @@
 #include "utest/utest.h"
 
 UTEST_MAIN();
+//testing idle
+UTEST(bsg, idle) {
+  Vehicle vehicle;
+  vehicle.velocity = 0;
+  vehicle.angleAccPedal = 0;
+  vehicle.angleBrakePedal = 0;
+  vehicle.requestCarStart = 0;
 
-UTEST(sum, exemple1) {
-  const uint16_t expectedResult = 5;
-  ASSERT_EQ(sum(2, 3), expectedResult);
+  BSG bsg;
+  bsg.currentMode = BSG_IDLE;
+  bsg.rpm = 0;
+  bsg.voltage = 0;
+  bsg.current = 0;
+
+  Engine engine;
+  engine.rpm = 0;
+
+  Battery battery;
+  battery.voltage = 0;
+  battery.current = 0;
+
+  const StateBSG bsgMode = selectBSGMode(vehicle, engine, bsg, battery);
+  ASSERT_EQ(bsgMode, BSG_IDLE);
+}
```

Comment:
Suggestion: Add line break between test cases to improve readability.

```suggestion
}

```